src/Associations.cc

```diff
@@ -265,61 +261,88 @@ void Associations::associateRefStars(double matchCutInArcSec, const Gtransfo *gt
         for (MeasuredStarIterator mi = catalog.begin(); mi != catalog.end();) {
             MeasuredStar &mstar = **mi;
 
-            auto fstar = mstar.getFittedStar();
-            if (!fstar) {
+            auto fittedStar = mstar.getFittedStar();
+            // measuredStar has no fittedStar: move on.
+            if (fittedStar == nullptr) {
                 ++mi;
                 continue;
             }
 
-            /*  keep FittedStar's which either have a minimum number of
-                measurements, or are matched to a RefStar
-            */
-            if (!fstar->getRefStar() && fstar->getMeasurementCount() < minMeasurements) {
-                auto f = std::const_pointer_cast<FittedStar>(fstar);
-                f->getMeasurementCount()--;
-                mi = catalog.erase(mi);
-            } else
+            // keep FittedStars which either have a minimum number of
+            // measurements, or are matched to a RefStar
+            if (!fittedStar->getRefStar() && fittedStar->getMeasurementCount() < minMeasurements) {
+                fittedStar->getMeasurementCount()--;
+                mi = catalog.erase(mi);  // mi now points to the next measuredStar.
+            } else {
                 ++mi;
+            }
         }  // end loop on objects in catalog
     }      // end loop on catalogs
 
-    /* now FittedStars with less than minMeasurements should have
-       zero measurementCount; */
-
+    // now FittedStars with less than minMeasurements should have zero measurementCount.
     for (FittedStarIterator fi = fittedStarList.begin(); fi != fittedStarList.end();) {
-        if ((*fi)->getMeasurementCount() == 0)
+        if ((*fi)->getMeasurementCount() == 0) {
             fi = fittedStarList.erase(fi);
-        else
+        } else {
             ++fi;
+        }
     }
 
     LOGLS_INFO(_log, "Fitted stars after measurement # cut: " << fittedStarList.size());
 }
 
+void Associations::normalizeFittedStars() const {
+    // Clear positions in order to take the average of the measuredStars.
+    for (auto &fittedStar : fittedStarList) {
+        fittedStar->x = 0.0;
+        fittedStar->y = 0.0;
+        fittedStar->setFlux(0.0);
+    }
+
+    // Iterate over measuredStars to add their values into their fittedStars
+    for (auto const &ccdImage : ccdImageList) {
+        const Gtransfo *toCommonTangentPlane = ccdImage->getPix2CommonTangentPlane();
+        MeasuredStarList &catalog = ccdImage->getCatalogForFit();
+        for (auto &mi : catalog) {
+            auto fittedStar = mi->getFittedStar();
+            assert(fittedStar != nullptr);  // This must be true, if selectFittedStars() was called.
+            auto point = toCommonTangentPlane->apply(*mi);
+            fittedStar->x += point.x;
+            fittedStar->y += point.y;
+            fittedStar->getFlux() += mi->getFlux();
+        }
+    }
+
+    // Finally, divide to get the average values that we summed above.
```

Comment:
Not sure comments like this are useful, especially since you already used the word "average" above.